### PR TITLE
Implement charm-info and charm-missing counters.

### PR DESCRIPTION
**Pull request by frankban**
Those stats counter are used in the legacy API.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/230/

File: internal/legacy/api.go [hunk 1]
```diff
@@ -100,15 +108,21 @@ func (h *Handler) serveCharmInfo(w http.ResponseWriter, req *http.Request) (inte
...
 			c.Sha256 = entity.BlobHash256
 			c.Revision = curl.Revision
+			// TODO frankban 2014-09-15: log possible IncCounter errors.
+			go h.store.IncCounter(charmStatsKey(curl, params.StatsCharmInfo))
 			// TODO(rog) include Digest if it exists in extra-info ?
 		} else {
-			// TODO(rog) increment charm-missing stats counter?
 			c.Errors = append(c.Errors, err.Error())
+			if curl != nil {
+				// TODO frankban 2014-09-15: log possible IncCounter errors.
```
Comment (rogpeppe) [inline internal/legacy/api.go]: Perhaps just put a TODO inside store.IncCounter and lose all the TODOs from its callers? Or perhaps add an IncCounterAsync method that does the "go" and could be changed to log in the future?

**Comment by rogpeppe** [comment]
LGTM with one trivial suggestion, thanks.

File: internal/legacy/api.go (hunk 1 above)
Comment (frankban) [inline internal/legacy/api.go]: Good idea (IncCounterAsync). Done.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/231/

**Comment by jrwren** [comment]
:+1:

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**